James-Lycett/workplace-wellness
# AdminProgressCharts

I explained this already via slack but here's a rundown

_This pull request also includes some of the stuff from the now closed #72 with the exception of the knexfile changes_

**To Test:**
- Use the Edit Goals button to change goals on the adminProgressCharts
- See that the radial changes to reflect the progress toward this new goal

**Changes:**
_Backend_
- Added lastMonthCompanyAverages to entries.service
- Removed a bunch of now unused functions from entries.service
- Removed useless /entries/all/metrics route from entries.router
- Renamed all of the calls for "lastMonthCompanyMetrics" to "lastMonthCompanyAverages"

_Frontend_
- Renamed all of the calls for "companyMetrics" to "companyAverages"
- AdminProgressCharts set the radials to the three agreed-upon stats (sleep_duration, stress_level, and daily_steps)
- AdminProgressCharts combined the three "calculate<something>Progress" functions into one
- AdminProgressCharts stored calculated progress in state so radial bar components would update when we change the goals
- AdminProgressCharts plugged in the now real numbers based on actual goals to the radial bar components
- Made the editGoal form input step attribute conditional so step=100 for daily steps
